### ISSUE TITLE
Naked Variable Interface

### ISSUE THREAD

**Issue by bnlawrence** [NCAS-CMS/pyfive#13]
We have determined that it would be desirable that we have a lightweight variable interface which we can pass to Dask, and potentially use for ActiveStorage initialisation within Dask as well.

It would need the following methods, and only these:
 - `shape`,
 - `__get_item__`
 - 'dtype`
 - `size`
 - `ndim`
 - `get_chunk_index`

To implement this we'd also need the filter pipeline and compression information as well as the b-tree cached in this object.

We need a route to get this from an h5netcdf object via it's `._h5ds` link to a `pyfive` object, so we need that method on our pyfive datasetobject. How we get to this is a moot point, do we get to it via a method on the h5netcdf Variable object (if so, we'd need to implement support for doing it with h5py objects as well ... ) or we could do it as a method which is in pyfive.

**Comment by bnlawrence** [comment]
Actually, it turns out that the obvious thing to do here is to make the new `h5d.DatasetID` class introduced as part of #4 cache all the relevant metadata. A partial implementation of that appears in [e32a1b0](https://github.com/NCAS-CMS/pyfive/commit/e32a1b03fad642d1005cd7764e0103edcad4f22e).

It is partial, because it's become apparent that implementing references properly means potentially iterating through file metadata using visititems  while trying to some dereferencing. The error manifests itself in the 
test `test_raise_error_invalid_dereference()`, which requires looking at the offset associated with the dataobjects of each item:
```
tests/test_high_level.py:205: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../mambaforge/envs/fesom/lib/python3.10/unittest/case.py:738: in assertRaises
    return context.handle('assertRaises', args, kwargs)
../../mambaforge/envs/fesom/lib/python3.10/unittest/case.py:201: in handle
    callable_obj(*args, **kwargs)
pyfive/high_level.py:56: in _dereference
    obj = self.file._get_object_by_address(ref.address_of_reference)
pyfive/high_level.py:221: in _get_object_by_address
    return self.visititems(
pyfive/high_level.py:147: in visititems
    ret = func(name, obj)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   lambda x, y: y if y._dataobjects.offset == obj_addr else None)
E   AttributeError: 'Dataset' object has no attribute '_dataobjects'

```
The `Dataset` used to keep it's dataobject in this manner, but it doesn't any longer, so as to break the need to go back to the file. The obvious solution is to keep a dummy dataset object here, which can deliver the offset address, but it's klunky.  For now we'll do that. But this note is by way of record why.

**Comment by bnlawrence** [comment]
This is addressed by [2d88101](https://github.com/NCAS-CMS/pyfive/commit/2d881014963a578ec5134828d0964163b7998e83).
